activerecord/lib/active_record/fixture_set/table_row.rb

```diff
@@ -33,6 +33,13 @@ def lhs_key
         def join_table
           @association.through_reflection.table_name
         end
+
+        def timestamp_column_names
+          ModelMetadata.new(@association.through_reflection.klass).timestamp_column_names
+        rescue StandardError
```

Comment:
without the exception handler here, this test crashes: https://github.com/ghiculescu/rails/blob/has-many-through-fixture-timestamps/activerecord/test/cases/fixtures_test.rb#L604

There isn't a `parrot_treasures` table in the [schema](https://github.com/ghiculescu/rails/blob/has-many-through-fixture-timestamps/activerecord/test/schema/schema.rb) so I'm not sure what the expected behavior is for that test. But I figure it's okay to be careful here. If someone's got a really weird setup, we won't populate timestamp columns, and then if they are required that will raise - which is what happens today anyway so it's not a regression.

Reply:
Add `pt.table_name = "parrots_treasures"` like here.

https://github.com/rails/rails/blob/63337e18605f365d621b5dca951f438080ec9e9a/activerecord/test/cases/fixtures_test.rb#L576

Reply:
Hmmm, if I do that, then the two tests are identical. I think the test names in https://github.com/rails/rails/commit/8e16a00de4f77f15e405d083d535b92a23779d9d are back to front, and the test works despite the table not working. I'll fix the test.

Reply:
This is the best way I could get the test passing without changing what the test is trying to test: https://github.com/rails/rails/pull/41533/files#diff-2a57b61bbf9ee2c23938fc571d403799f68b4b530d65e2cde219a429bbf10af5R761